Sources/Diagnostics/DiagnosticsTracker.swift

```diff
@@ -91,6 +98,24 @@ final class DiagnosticsTracker: DiagnosticsTrackerType {
         )
     }
 
+    func trackPurchaseRequest(wasSuccessful: Bool,
+                              storeKitVersion: StoreKitVersion,
+                              errorMessage: String?,
+                              errorCode: Int?,
+                              storeKitErrorDescription: String?) async {
+        await track(
+            DiagnosticsEvent(eventType: .applePurchaseAttempt,
+                             properties: [
+                                .successfulKey: AnyEncodable(wasSuccessful),
+                                .storeKitVersion: AnyEncodable("store_kit_\(storeKitVersion.debugDescription)"),
+                                .errorMessageKey: AnyEncodable(errorMessage),
+                                .errorCodeKey: AnyEncodable(errorCode),
+                                .skErrorDescriptionKey: AnyEncodable(storeKitErrorDescription)
```

Comment:
❤️ I think this could be useful! We probably should add it to the event I added in #4253 if we can.

Reply:
yes! I think it will be super useful. I thought about using an int code, but SK2 errors don't have an int representation. So I decided to go with the name of the event. It will make it easier to analyse as well.